static/app/components/profiling/transactionToProfileButton.tsx

```diff
@@ -2,26 +2,27 @@ import {Location} from 'history';
 
 import {Button, ButtonProps} from 'sentry/components/button';
 import {t} from 'sentry/locale';
+import {EventTransaction} from 'sentry/types';
 import {trackAnalytics} from 'sentry/utils/analytics';
 import {generateProfileFlamechartRouteWithQuery} from 'sentry/utils/profiling/routes';
 import useOrganization from 'sentry/utils/useOrganization';
 
-import {useTransactionProfileId} from './transactionProfileIdProvider';
-
 interface Props {
+  event: EventTransaction;
   projectSlug: string;
   children?: React.ReactNode;
   query?: Location['query'];
   size?: ButtonProps['size'];
 }
 
 function TransactionToProfileButton({
+  event,
   projectSlug,
   query,
   children = t('View Profile'),
   size = 'sm',
 }: Props) {
-  const profileId = useTransactionProfileId();
```

Comment:
Can you clarify why this hook doesn't work anymore? Is it a particular use-case that was introduced in this PR?

Reply:
Hey, there's isn't really a need to be making that extra call to the profiling endpoint for the id here. Everywhere this component is used we have access to the event and thus the profile id.

Reply:
This was here due to legacy reasons since when profiling was originally introduced, the profile id was not available on the transaction. Thus the only way to get the corresponding profile id was to query profiles with the transaction id.

At this time, this is no longer true so it is safe to assume that directly read the profile id from the profile context like you're doing here.

Reply:
Isn't it the context provider that actually queries for the transaction ID? This hook is just the consumer. If we want to go with this change I think we should clean up the providers too

Reply:
Let's follow up this PR with another one to remove the extra providers if we think it's unnecessary/dead code